### PR TITLE
Add sv1 client and server test

### PR DESCRIPTION
**Pull request by Fi3**

### CODE REVIEW

File: examples/sv1-client-and-server/src/main.rs [hunk 1]
```diff
@@ -559,6 +540,22 @@ async fn initialize_client(client: Arc<Mutex<Client>>) {
 }
 
 fn main() {
+    match get_runtime() {
+        Some(s) => {
+            std::thread::spawn(move || {
+                let now = std::time::SystemTime::now();
+                loop {
+                    std::thread::sleep_ms(100);
```
Comment (darricksee) [inline examples/sv1-client-and-server/src/main.rs:548]: Did you mean to sleep for 1s?

**Review by darricksee — APPROVED** [review]

File: .github/workflows/test.yaml [hunk 2]
```diff
@@ -66,3 +66,9 @@ jobs:
         with:
           command: run
           args: --bin ping_pong_without_noise -- 10
+
+      - name: Run sv1-client-and-server example
+        uses: actions-rs/cargo@v1
+        with:
+          command: run
+          args: --bin client_and_server -- 10
```
Comment (darricksee) [inline .github/workflows/test.yaml:74]: I would bump this up to 30s at least - it's odd that in 10s it's only doing this much:

```     Running `target/debug/client_and_server 10`
Server listening on: 127.0.0.1:44691
SystemTime { tv_sec: 1663600688, tv_nsec: 682836888 }-CLIENT - connected to server at 127.0.0.1:44691
SERVER - Accepting from: 127.0.0.1:50310
SERVER - message: {"id":"1663600688682940388","method":"mining.configure","parameters":[["version-rolling"],{}]}
Test Success - ran for 11 seconds```

**Comment by rrybarczyk** [comment]
@Fi3 i have been making some changes to the sv1-protocol #201 which impacts the sv1-client-and-server example. Not sure if you want me to merge these changes in before we merge in the tests? It shouldn't impact it too much, so I can always rebase them with this branch or with main once you push this branch

**Comment by darricksee** [comment]
> @Fi3 i have been making some changes to the sv1-protocol #201 which impacts the sv1-client-and-server example. Not sure if you want me to merge these changes in before we merge in the tests? It shouldn't impact it too much, so I can always rebase them with this branch or with main once you push this branch

I'm not sure we need to merge these in at all at this point. This doesn't address the deadlock issue which I'm tracking down now. I'm hoping to have the root cause figured out soon then we can decline this pr.

**Comment by darricksee** [comment]
I'm pretty sure this PR can be cancelled/removed as the client/server example is now running in main.

**Pull request closed without merging**